godot-ffi/src/binding/single_threaded.rs

```diff
@@ -30,13 +38,53 @@ impl BindingStorage {
     #[inline(always)]
     unsafe fn storage() -> &'static Self {
         static BINDING: BindingStorage = BindingStorage {
+            #[cfg(wasm_nothreads)]
+            initialized: Cell::new(false),
+
+            #[cfg(not(wasm_nothreads))]
             main_thread_id: Cell::new(None),
             binding: ManualInitCell::new(),
         };
 
         &BINDING
     }
 
+    /// Marks the binding storage as initialized or deinitialized.
+    /// We store the thread ID to ensure future accesses to the binding only come from the main thread.
+    ///
+    /// # Safety
+    /// Must be called from the main thread. Additionally, the binding storage must be initialized immediately
+    /// after this function if `initialized` is `true`, or deinitialized if it is `false`.
+    ///
+    /// # Panics
+    /// If attempting to deinitialize before initializing, or vice-versa.
+    unsafe fn set_initialized(&self, initialized: bool) {
+        #[cfg(wasm_nothreads)]
+        {
+            if !initialized && !self.initialized.get() {
+                panic!("deinitialize without prior initialize");
+            }
+
+            // 'std::thread::current()' fails when linking to a Godot web build without threads. When compiling to wasm-nothreads,
+            // we assume it is impossible to have multi-threading, so checking if we are in the main thread is not needed.
+            // Therefore, we don't store the thread ID, but rather just whether initialization already occurred.
+            self.initialized.set(initialized);
+        }
+
+        #[cfg(not(wasm_nothreads))]
+        {
+            if initialized {
+                self.main_thread_id.set(Some(std::thread::current().id()));
+            } else {
+                self.main_thread_id
+                    .get()
+                    .expect("deinitialize without prior initialize");
+
+                self.main_thread_id.set(None);
+            }
+        }
+    }
```

Comment:
With the above changes, this can then be:

```suggestion
        #[cfg(not(wasm_nothreads))]
        {
            let thread_id = initialized.then(|| std::thread::current().id());
            self.main_thread_id.set(thread_id);
        }
    }
```

Same for the other block.